core/src/avm2/object/class_object.rs

```diff
@@ -911,17 +912,35 @@ impl<'gc> TObject<'gc> for ClassObject<'gc> {
     fn apply(
         &self,
         activation: &mut Activation<'_, 'gc>,
-        nullable_param: Value<'gc>,
+        nullable_params: &[Value<'gc>],
     ) -> Result<ClassObject<'gc>, Error<'gc>> {
         let self_class = self.inner_class_definition();
 
         if !self_class.read().is_generic() {
-            return Err(format!("Class {:?} is not generic", self_class.read().name()).into());
+            return Err(make_error_1127(activation));
+        }
+
+        if nullable_params.len() != 1 {
+            let class_name = self
+                .inner_class_definition()
```

Comment:
not necessarily in this PR, but maybe we should refactor these into `self.name()`